pkg/azure/provider/helpers/driver.go

```diff
@@ -686,9 +785,18 @@ func getDataDisks(specDataDisks []api.AzureDataDisk, vmName string) []*armcomput
 			},
 			Name: to.Ptr(dataDiskName),
 		}
+		if specDataDisk.ImageRef != nil {
+			diskID := imageRefDiskIDs[DataDiskLun(specDataDisk.Lun)]
+			if diskID == nil {
```

Comment:
is this even possible?
The reason is that we operate on the same input which is api.AzureDataDisk. So unless there is a coding mistake when constructing  imageRefDiskIDs map[DataDiskLun]DiskID this cannot happen, right?
If you are just adding this check as a defence mechanism then the error should categorically state "unexpected error, this cannot happen and points to a coding error". This makes it very clear to whosoever is looking at the logs.